### PR TITLE
Add support for axis names in jax.scipy.special.logsumexp

### PR DESCRIPTION
**Pull request by apaszke**

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
LGTM!

Will we need a `pbroadcast` call here when `keepdims=True` and reducing over a named axis? If so we might want to leave a comment or something for our future selves.

**Comment by apaszke** [comment]
I replied in chat. We don't need any `pbroadcast`s here! `expand_dims` is only necessary to avoid shape errors in positional broadcasting, because removing dimensions can cause misalignment. But there's no misalignment when you have names!

**Comment by mattjj** [comment]
Ah of course, you're right! It just keeps singleton dimensions, because positional axes are bad.
